tl

```diff
@@ -863,14 +970,18 @@ if cmd == "build" then
          check_parent_dirs(project.source_file_map[fname])
       end
    end
-   for path in project:files(inc_patterns, exc_patterns, tlconfig["source_dir"]) do
+   for path,is_build in project:file_with_is_build(inc_patterns, exc_patterns, tlconfig["source_dir"]) do
       --TODO: make this better
       local valid = true
       if not (path:match("%.tl$") and not path:match("%.d%.tl$")) then
          valid = false
       end
       if valid then
-         project.source_file_map[path] = path:gsub("%.tl$", ".lua")
+         local work_on = path:gsub("%.tl$", ".lua")
+         if is_build then
+            work_on = work_on:gsub(build_path,"")
```

Comment:
I think you want `remove_leading_path(build_path, work_on)` here as this `gsub` turns the path into an absolute one `foo/bar` -> `/bar`